game-core/src/main/java/games/strategy/debug/error/reporting/ConfirmationDialogController.java

```diff
@@ -0,0 +1,86 @@
+package games.strategy.debug.error.reporting;
+
+import java.net.URI;
+
+import javax.swing.JEditorPane;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.HyperlinkEvent;
+
+import org.triplea.http.client.ServiceResponse;
+import org.triplea.http.client.error.report.create.ErrorReportResponse;
+
+import games.strategy.net.OpenFileUtility;
+import games.strategy.triplea.UrlConstants;
+import swinglib.JPanelBuilder;
+
+/**
+ * This controller is responsible for showing success/failure confirmation dialogs
+ * after an error report has been submitted.
+ */
+// Not static to allow for mock verifications.
+@SuppressWarnings("MethodMayBeStatic")
```

Comment:
Being forced to change the implementation in order to test (e.g. removing `final` from a class, changing class methods to instance methods just for mocking, etc.) is kind of a smell.  Having to make those types of changes is usually indicative that an interface should be extracted.

Reply:
222ceef51